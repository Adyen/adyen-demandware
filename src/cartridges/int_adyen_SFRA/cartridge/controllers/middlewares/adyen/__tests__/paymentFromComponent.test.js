/* eslint-disable global-require */
let paymentFromComponent;
let res;
let req;

beforeEach(() => {
  const { adyen } = require('../../index');
  paymentFromComponent = adyen.paymentFromComponent;
  jest.clearAllMocks();
  req = { form: { data: { paymentMethod: { type: 'mocked_type' } } } };
  res = { json: jest.fn() };
});

afterEach(() => {
  jest.resetModules();
});

describe('Payment from Component', () => {
  it('should cancel transaction', () => {
    req.form.data.cancelTransaction = true;
    req.form.data = JSON.stringify(req.form.data);
    const expected = paymentFromComponent(req, res, jest.fn());
    expect(expected).toEqual({});
  });
  it('should return json response', () => {
    req.form.data = JSON.stringify(req.form.data);
    paymentFromComponent(req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
});
