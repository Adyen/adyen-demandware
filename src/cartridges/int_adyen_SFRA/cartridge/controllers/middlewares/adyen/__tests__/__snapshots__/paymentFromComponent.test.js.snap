// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment from Component should return json response 1`] = `
Array [
  Array [
    Object {
      "resultCode": "Authorised",
    },
  ],
]
`;
